sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4200308&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4200308&tab=artifacts#/) on master @ [236797939ac41bb759feee9bb676a5d6ca381237](https://github.com/cockroachdb/cockroach/commits/236797939ac41bb759feee9bb676a5d6ca381237):


```
            │
            └── • filter
                │ filter: st_coveredby('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlF9v2jwUxu_fT2Gdm4JeF2wndKuv6J90y0ShC0xbNaMqxWddVIgz21RUFd99Cuna0opQfGE4x_6dk-eR7Qdwf6YgIfpx0TuK-6RxGg9Hw6-9JhlGvehkRG7JWTI4Jzdornx6PUXy_XOURMT5q4m5Q4v6-r6xd_6tN4ovBnF_1GiIliCiJZqUNIIWI0GLNZt7Un6KBufRKLmkZalZkwyS0yghx5fkFijkRmM_naED-RM4UBAwplBYM0HnjC3TD6tNsV6AZBSyvJj7Mj2mMDEWQT6Az_wUQULf7Jui3QEKGn2aTVfblhTM3D9Dzqc3CPJgSV8U5vWFR6X-BFONts3WysOTPd1S3VWWa1wAhRMznc9yJ8ltJRsoDIu0TLQVHCu1-KWVWnCm1IJtm2B_V4YrIGmuScCI8b_ROthkA9_Fhji_Q-tRn2VTjxZtm6978W89WhSWmJx0uSSuVE2cT62XKxXBh45STDClGNs2AcFc74qV4l-ppzCYe0m6fKMPYhcfvpgsfzwNYtNpKGw2S-39c2vaFRu7B7t0f3I_WO9d5eWrC8o4C1k1xOMvZ7z6c3h0-DQYD9_EzzvXxsfwTbwnX97zrmi-w_JwF9FDYz3adrguucv_31i-s1Z-y8uRoCtM7vBdTwdbjimgvsHqdXJmbid4Yc1k1aYKBytuldDofLV6UAVxXi2VH_gS5rWwqIdFLRzUw0EtHNbDYS3cqYc7tTB7BY-X__0NAAD__xVlF_8=
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • sort
            │ order: +k
            │
            └── • filter
                │ filter: st_coveredby('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlN9P2zAQx9_3V1j3Qqt5re0Ebfip_AhbpkJZ2mlDc4VCfWMRbdzZLgpC_d-nJAwoqIH4we3d-XN335OdO3B_5yAh-nk23I9PSecoHk_G34ZdMo6G0eGEXJPjZHRCrtBc-PRyjuTHlyiJiPMXM3ODFvXlbWfn5PtwEp-N4tNJpyN6goie6FLSCXqMBD3W7e5I-TkanUST5JyWqRZdMkqOooQcnJNroJAbjafpAh3IX8BhSmFpzQydM7Z03VUHYl2AZBSyfLnypXtKYWYsgrwDn_k5goRJ2WKCqUbbZ0BBo0-zeZX2QcGgbOAiyzUWQOHQzFeL3ElyXXcGFMbLtHT0FRwoVfzWShWcKVWw1zb40JbhCkiaaxIwYvwftA6mawpm5R8lOp9eIUi-pm8fQ5zfoPWoj7O5R4u2zzdn8T8eFUtLTE4GXBJXqibOp9bLSkXwcVcpJphSjL22AcFct8VK8c_UUxitvCQDvnUOos0cvposv78NYtttWNpskdrbx9J0ILZWD9pUf5h-sFm79stnb4hxFrJ6iftfznj9Z29_72ExHr6wH09urE_hC3tHPn2KA9F9w8jDNqLHxnq0_XBT8oC_35p-t036BN3S5A430m_LzNZTCqivsP6IOLOyMzyzZlaVqc1RxVUOjc7XUV4bcV6HygafwrwRFs2waISDZjhohMNmOGyEd5_B0_W7fwEAAP__bbfl1A==
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist with config 5node: 14 tests, 1 failures
    logic.go:3379: 
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:238: error while processing
    logic.go:3379: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:238: too many errors encountered, skipping the rest of the input
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join with config 5node: 98 tests, 0 failures
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_json_array_dist with config 5node: 30 tests, 0 failures
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (3.81s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist
+++ b/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist
@@ -208,7 +208,7 @@ Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlN9v2jAQx9_3V
 #
 # TODO(treilly): What the text above claims does not square with the figure
 # generated below.
-query T
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE ST_CoveredBy('MULTIPOINT((2.2 2.2), (3.0 3.0))'::geometry, geom) ORDER BY k
 ----

